client/rpcserver/handlers.go

```diff
@@ -452,4 +461,71 @@ Registration is complete after the fee transaction has been confirmed.`,
 		returns: `Returns:
     string: The message "` + fmt.Sprintf(feePaidStr, "[fee]") + `"`,
 	},
+	exchangesRoute: {
+		pwArgsShort: ``,
+		argsShort:   ``,
+		cmdSummary:  `List assets supported by all registered DEX servers their and active trades.`,
+		pwArgsLong:  ``,
+		argsLong:    ``,
+		returns: `Returns:
+    map: The exchanges result.
+    {
+      "[DEX url] (map): {
+        "url" (string): The DEX URL.
```

Comment:
I think the result is a bit repetitive:

```json
$   ./dexcctl -u u -P p exchanges
{
  "https://dex-test.ssgen.io:7232": {
    "url": "https://dex-test.ssgen.io:7232",
    "markets": {
      "btc_ltc": {
        "name": "btc_ltc",
        "baseid": 0,
        "basesymbol": "btc",
        "quoteid": 2,
        "quotesymbol": "ltc",
        "epochlen": 6000,
        "startepoch": 264696851,
        "buybuffer": 1.25,
        "orders": null
      },
      "dcr_btc": {
        "name": "dcr_btc",
        "baseid": 42,
        "basesymbol": "dcr",
        "quoteid": 0,
        "quotesymbol": "btc",
        "epochlen": 10000,
        "startepoch": 158818111,
        "buybuffer": 1.25,
        "orders": [
          {
            "dex": "https://dex-test.ssgen.io:7232",
            "market": "dcr_btc",
            "type": 1,

```

Either this could be an array of exchange objects (`[ unkeyed exchange objects ]`), or a JSON object keyed by dex url/host but without the repeated `"url"` in the exchange object.  Also the market names are repeated so the same thing could be applied (`"markets": [ market objects ]` or stick with an object with market name keys but no "name").

Also note that each item in `"orders"` repeats the `"dex"` and `"market"` that are implied by the parent object.

If these changes require a ton of work and/or new types, I won't push it.

Reply:
This is now selectively removing the mentioned fields from the response.